march_hardware/src/imotioncube/imotioncube.cpp

```diff
@@ -399,9 +399,9 @@ uint32_t IMotionCube::computeSWCheckSum(int& start_address, int& end_address)
       start_address = std::stoi(token, nullptr, 16);
       token = "0";
     }
-    if (pos - old_pos < 2)  // delimiter has length of 1 two \n in a row has difference in positions of 1
+    if (pos == old_pos)
     {
-      end_address = end_address + start_address - 2;  // The -2 compensates the offset of the end_address
+      end_address = end_address + start_address - 1;  // The -1 compensates the offset of the end_address
```

Comment:
To make this completely safe, you should also change line 409. Line 402 assumes that the delimiter has been completely skipped.
```diff
- old_pos = pos + 1;
+ old_pos = pos + delimiter.length();
```